component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -68,6 +68,7 @@ public PrivateKeyJWTClientAuthenticator() {
         int rejectBeforePeriod = DEFAULT_VALIDITY_PERIOD_IN_MINUTES;
         boolean preventTokenReuse = true;
         String tokenEPAlias = DEFAULT_AUDIENCE;
+        List<String> validAudiences = new ArrayList<>();
```

Comment:
Is the authenticator a method level when it obtain the lists of valid audiences? If not for concurrent requests will we not end up with ciba audiences for general (non-ciba) request?

Reply:
This is fixed with https://github.com/wso2-extensions/identity-oauth-addons/pull/88/commits/09259253f54f66af8b0b88ff2cc7feb57e770c2c

Only the first element of audience array(issuer Identifier) will be used if its a non CIBA call

Reply:
Correction: By constructor, one can setup custom audience values for their JWT authenticator. In the default PkJWT authenticator, validAudiences is an empty string therefore this class level variable is not used. Instead, a class level variable is used which returns issuer identifier on a non ciba call, and all issuer identifier, ciba endpoint and token endpoint on a ciba call